### ISSUE TITLE
Homestead issues SARs after failing to contact Cassandra

### ISSUE THREAD

**Issue by graemerobertson** [Metaswitch/homestead#431]
#### Symptoms
I hit https://github.com/Metaswitch/homestead/issues/387, and whilst looking at the SAS trace (attached), I noticed that after Homestead failed to contact Cassandra, it issued a SAR anyway.

#### Impact
There was no observable impact in this particular flow but I'm concerned about the following two hypothetical scenarios...
* We fail to find a registered subscriber's details in Cassandra on the terminating side of a call (due to a Cassandra failure) and we send a SAR to the HSS with Server-Assignment-Type UNREGISTERED_USER which incorrectly moves the subscriber into unregistered state (and then they won't be able to make calls without re-registering).
* We fail to find a subscriber's details in Cassandra on the terminating side of a call (due to a Cassandra failure) and we send a SAR to the HSS which successfully changes the subscriber's state (e.g. a deregistration?). However, the request from Sprout timed out so Sprout has queried another Homestead node, and triggered another SAR, which has failed because the subscriber has already changed state (e.g. a deregistration, this time we get back an error indicating the subscriber is not registered), and this failure gets propagated back to the user.

In short, I'm not sure we should be sending a SAR when we fail to contact Cassandra, especially when we've hit https://github.com/Metaswitch/homestead/issues/387 given that the THRIFT_EAGAIN error appears to take so long that the request from Sprout will timeout and so the response to this SAR will be ignored by Sprout anyway.

(This issue may not just be restricted to the SAR flows - I haven't thought about it too much)

#### Release and environment 

The sample SAS trace is from CC-homestead-10.0.0-170221-~255.00.0.ova

#### Steps to reproduce

No steps to repro per se, given this is fairly hypothetical

**Comment by graemerobertson** [comment]
I don't think this is critical (at least whilst we're happy to live with https://github.com/Metaswitch/homestead/issues/387), but I've set to high priority because it might be worth somebody thinking about whether there are any more serious implications of this potential issue. Feel free to change the priority.
